README rewrite for v2

jlengstorf opened the pull request:
This is the beginnings of a rewrite for the v2 README.

Huge thanks to https://github.com/choojs/choo and https://github.com/zeit/next.js for creating excellent READMEs I could steal liberally from. 😅 

### [See the new work-in-progress README](https://github.com/jlengstorf/gatsby/blob/docs/6569-readme-rewrite/README.md)

## Remaining work

- [x] Add a “what is Gatsby and why should I care?” section (see [the React README](https://github.com/facebook/react/#react------) for a good example)
- [x] Decide if we should keep the “advanced” section
	- How do we keep this from becoming a ton of duplicate doc maintenance work?
	- If we _don’t_ keep it, how do we make it easy for advanced users to find a quick reference of advanced topics?
	- If we _do_ keep it, what should be included?
- [x] Decide on pointing to v1 vs. v2 docs
- [x] Add a link to the Code of Conduct in the Contributing section
- [x] Add a license callout at the bottom of the README
- [x] Decide whether or not to keep Open Collective
	- Answered: [we should not](https://github.com/gatsbyjs/gatsby/pull/6843#pullrequestreview-141334559)


Overall I feel pretty good about this. Looking forward to feedback!

cc: @gatsbyjs/inkteam

KyleAMathews commented:
Deploy preview for *using-postcss-sass* failed.

Built with commit 8973de78a89cc9c773d9e4c8656d341c994273a0

https://app.netlify.com/sites/using-postcss-sass/deploys/5b5cf651b13fb1644cf6cda1

KyleAMathews commented:
Deploy preview for *gatsbyjs* failed.

Built with commit 8973de78a89cc9c773d9e4c8656d341c994273a0

https://app.netlify.com/sites/gatsbyjs/deploys/5b5cf650b13fb1644cf6cd99

KyleAMathews commented on README.md:
````diff
@@ -1,371 +1,217 @@
+    Get your Gatsby blog set up in a single command:
+
+    ```sh
+    npx gatsby-cli new my-blazing-fast-site https://github.com/gatsbyjs/gatsby-starter-default#v2
````
Also until we release v2, let's keep pushing people to use the default version.

giraffesyo replied:
I don't think this should be cherry-picked to the `v1` branch. Instead, this should stay only on the `master` branch and content within should remain relevant to v2. This makes the most sense to me since v2 is already the `master` branch of the repository.

KyleAMathews commented on README.md:
```diff
@@ -1,371 +1,217 @@
+  <span> · </span>
+  <a href="https://next.gatsbyjs.org/plugins/">Plugins</a>
+  <span> · </span>
+  <a href="https://next.gatsbyjs.org/docs/gatsby-starters/">Starters</a>
```
Maybe add a link to the showcase here to duplicate the site's main menu?

KyleAMathews commented on README.md:
```diff
@@ -1,371 +1,217 @@
+
+    Your site is now running at `http://localhost:8000`. Open the the `my-blazing-fast-site` directory in your code editor of choice and edit `src/pages/index.js`. Save your changes and the browser will update in real time!
+
+At this point, you’ve got a fully functional Gatsby website. For additional information on how you can customize your Gatsby site, see our [plugins](https://next.gatsbyjs.org/plugins/) and [the official tutorial](https://next.gatsbyjs.org/tutorial/).
```
Let's not link people to next.gatsbyjs.org — it's a temporary thing and we'll be deleting it soon.

KyleAMathews replied:
@jlengstorf and I talked offline and I'm on board to linking to v2 stuff from the README

jlengstorf commented:
Updated per @KyleAMathews’s feedback. Also edited the main PR text to include a checklist of remaining work.

For the advanced section, I think there are arguments both ways:

-	**PRO:** For an advanced developer, a GitHub repo is a more familiar way of gathering a quick overview of a repo. 
-	**CON:** It’s difficult to know _which_ concepts are actually useful since the breadth of what Gatsby can do is pretty wide.
-	**CON:** High potential for duplicate examples, creating heavier maintenance costs as things evolve.

A potential alternative option is to borrow the structure of the [React README’s “Documentation” section](https://github.com/facebook/react/#documentation).

@shannonbux I’d be curious to hear your thoughts on this.

jlengstorf commented:
Thanks for the link fix, @giraffesyo!

More updates:

1. Added an overview of Gatsby’s key features and why they matter.
2. Replaced the “advanced” section with an improved docs overview.
3. Added the Code of Conduct to the contributing section.

I more or less just stole ideas from React, so props to their team for writing a good README.

I think this addresses the issues with maintaining an “advanced” section in two places: rather than trying to decide what the most important code samples are and keeping them up to date in two places, we can just call out where the advanced code samples live and link to them.

(@shannonbux, this does bring up a potential issue with the docs, though: we can't directly link to a section in the docs since the sections are just accordions, not actual links. Is that worth addressing?)

I also decided to skip the license callout at the bottom since we have the badge at the top. Figured the duplication wasn't worth the added length.

Assuming @KyleAMathews et al. agree with me about linking to next.gatsbyjs.org from `master` until we hit v2 stable, I _think_ this is pretty close to done.

giraffesyo commented:
Great work on the new README file. It looks really good.

I think that there should also be a section about what someone should do to update v1 (e.g. bug fixes) and a note about how long it will be supported.

jlengstorf commented:
@giraffesyo good call. Added.

jlengstorf commented:
Great! Should we merge it? Can someone add an approval so we can get it in?

m-allanson commented:
😍 nice rewrite @jlengstorf!

shannonbux commented:
Hey @jlengstorf this is looking great--super great! Just a bit of feedback (had meetings all day until now!):
- I got confused when I read this phrase:
> We are currently only accepting bug fixes for Gatsby v1. All new features, docs enhancements, and other improvements must be created against v2 (the master branch).

Does it mean: "We accept bug fixes, new feature requests, docs enhancements, and other improvements against v2 (the master branch). For v1, we accept only bug fixes."

Not sure that's clearer, just wanted to make sure it's clear that bug fixes are also accepted for v2.

- It seems like there are a lot of repeated hyperlinks in the doc w/ different wording. Don't think it's an emergency, just wondering if there's a way to streamline so people know for sure what to click on and don't open multiple tabs of the same thing.

jlengstorf commented:
@shannonbux For the v1 section, since it's under a "Contributing to v1" heading I _think_ it should avoid any confusion about what can be contributed to v2. We could potentially even strip it down further, I guess:

> ### Contributing to Gatsby v1
> We are currently only accepting bug fixes for Gatsby v1.

For the repeated hyperlinks, I'm not sure what the right approach is there. My preference is to link to important ideas whenever they're mentioned, and with alternative phrasing to try and ensure that even if the first description doesn't alert the reader that it's the answer to their question, maybe the alternative phrasing will.

I don't have super strong feelings about it, honestly, so I'm happy to look at revisions if you want to tweak this. 😄

calcsam commented:
@jlengstorf I like the stripping it down even further idea.

shannonbux commented:
Ok stripping down looks good, and we can keep the hyperlinks in the absence
of other arguments (I only really know that for beginners who are opening
every hyperlink in a new tab, it'd get confusing. However, more expert ppl
probably won't have a problem).

On Mon, Jul 30, 2018 at 9:07 PM, Sam Bhagwat <notifications@github.com>
wrote:

> @jlengstorf <https://github.com/jlengstorf> I like the stripping it down
> even further idea.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gatsbyjs/gatsby/pull/6843#issuecomment-409079463>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/Ae9o2ljigXDUS9QSGpzjSuEsJvQdkQykks5uL8nXgaJpZM4VlNtF>
> .
>
